plugins/cache-portals-on-map.js

```diff
@@ -29,11 +29,60 @@ window.plugin.cachePortalDetailsOnMap.entityInject = function(data) {
   }
   data.callback(ents);
 };
+/****************************************************************************************/
+/* extend portal's GUID lookup to search in cached portals as well                      */
 
+window.cachePortalDetailsOnMap.findPortalGuidByCacheE6 = function(latE6, lngE6) {
+  let guid = window.cachePortalDetailsOnMap.findPortalGuidByPositionE6old(latE6, lngE6);
+  //PLUGIN.log("findPortalGuidByPositionE6old %s %s -> %s", latE6, lngE6, guid);
+
+  // PLUGIN: cache > cache
+  if (guid == null) {
+    const cache = window.plugin.cachePortalDetailsOnMap;
+    if (cache && cache.cache) {
+      for (let g in cache.cache) {
+        let p = cache.cache[g];
+        if (latE6 == p.ent[2][2] && lngE6 == p.ent[2][3]) {
+          PLUGIN.log("newFindPortalGuidByPositionE6 matched in cache! %s %s -> [%s] = %o", latE6, lngE6, g, p);
```

Comment:
» error: ReferenceError: PLUGIN is not defined
🤔
Anyway, it seems that this code will be called very often with a large number of portals?
I suggest replacing it with
`// console.log("newFindPortalGuidByPositionE6 matched in cache! %s %s -> [%s] = %o", latE6, lngE6, g, p);`

Reply:
CR implemented